### ISSUE TITLE
`SVGElement.className` should be read-only

### ISSUE THREAD

**Issue by inad9300** [Microsoft/TypeScript#19548]
When trying to assign a value to it, the browser yells: `TypeError: Cannot assign to read only property 'className' of object '#<SVGAElement>'`.

Furthermore, the type of the `className` property is set as `any`, but it should be `SVGAnimatedString`.

**Comment by DanielRosenwasser** [comment]
Right now the `browser.webidl.xml` specifies `readonly` and `SVGAnimatedString`, so I'm not sure if this is a synchronization issue or not.

https://github.com/Microsoft/TSJS-lib-generator/blob/5f57850e588937aeddfa702e85008456178313ed/inputfiles/browser.webidl.xml#L9324

**Comment by mhegazy** [comment]
PRs welcomed. You can find more information about contributing lib.d.ts fixes at https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md#contributing-libdts-fixes.

**Comment by a-tarasyuk** [comment]
@mhegazy @DanielRosenwasser  [SVGElement#className](https://svgwg.org/svg2-draft/types.html#__svg__SVGElement__className)

**Comment by mhegazy** [comment]
thanks @a-tarasyuk. closing since the attribute is already deprecated.

**Comment by inad9300** [comment]
I don't think this is the right move. This should be followed up and the property should be properly marked as deprecated. (What *properly* means is something the TypeScript team must figure out as well, as there are currently different ways of indicating so: we have `Document.bgColor` with a comment saying "Deprecated", `ElementTagNameMap` with the `@deprecated` JSDoc tag, and `DOML2DeprecatedColorProperty` carrying this information on its name.)

Most importantly, suppressing the type information with `any` is completely misleading and can only lead to problems. Now, according to TypeScript, the following code is valid:

```ts
function f(e: Element) {
    e.className = 'xyz'
}

f(document.createElementNS('http://www.w3.org/2000/svg', 'a'))
```

I find this kind of mistake easy to make, and the code will completely fail to change the class of any SVG element. MDN's documentation, the reference for so many developers, also doesn't help here unfortunately, as it mentions `className` only as part of [`Element`](https://developer.mozilla.org/en-US/docs/Web/API/Element#Properties), and neither as part of [`HTMLElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement#Properties) nor [`SVGElement`](https://developer.mozilla.org/en-US/docs/Web/API/SVGElement#Properties), so according to them `SVGElement.className` is just a string.

On top of that, despite the deprecation, browsers will still support this property for years. So please, set the proper type, and mark the property as deprecated.

**Comment by RyanCavanaugh** [comment]
className is now marked `deprecated`. I don't think fiddling with the type would be the best thing.
